pkg/mtls/override_parser_release.go

```diff
@@ -0,0 +1,8 @@
+//go:build release
+
+package mtls
+
+// Override does not do anything in release builds
+func (c *certificateParserWrapper) Override(_ CertificateParser) {
+	log.Warn("Override called in production code")
```

Comment:
```suggestion
	log.Warn("Override certificate parser must not be called in production code")
```

Reply:
Add some more context to the warning message

Reply:
Done